src/js/classes/Graph.js

```diff
@@ -0,0 +1,375 @@
+    yHeader.appendChild(document.createTextNode(this.data[0].yHeader));
+    headerRow.appendChild(filler);
+    headerRow.appendChild(xHeader);
+    headerRow.appendChild(yHeader);
+    thead.appendChild(headerRow);
+    table.appendChild(thead);
+
+    // render cells from data
+    const tbody = document.createElement("tbody");
+    this.data.forEach(data => {
+      const row = document.createElement("tr");
+      const name = document.createElement("td");
+      name.className = "scatter-tooltip-name";
+      name.appendChild(document.createTextNode(data.name));
+      const xText = document.createElement("td");
+      xText.appendChild(document.createTextNode(data.xText));
+      const yText = document.createElement("td");
+      yText.appendChild(document.createTextNode(data.yText));
+      row.appendChild(name);
+      row.appendChild(xText);
+      row.appendChild(yText);
+      tbody.appendChild(row);
+    });
+    table.appendChild(tbody);
+    tooltip.appendChild(table);
+
+    // set tooltip placement based on first point
+    if (!this.tooltipFixed) {
+      tooltip.style.top = this.ys[0];
+      tooltip.style.left = this.xs[0];
+    }
+    this.tooltip = tooltip;
+    this.container.appendChild(this.tooltip);
+  }
+}
+
+
+export class ScatterPlot {
+  constructor(data, xAxis, yAxis, toText, container) {
+    this.data = data;
+    this.xAxis = xAxis;
+    this.yAxis = yAxis;
+    this.ticks = {"x": [], "y": []};
+    this.axisLabels = {"x": [], "y": []};
+    this.toText = toText;
+    this.container = container;
+    this.plotContainer = this.container.getElementsByClassName("plot-container")[0];
+    this.plot = this.container.getElementsByClassName("scatter-plot")[0];
+    this.points = this.createPoints();
+    this.mobileSizing = window.innerWidth < 670;
+    this.showOutliers = false;
+    this.setUpSearchBar();
+    this.setUpOutlierButton();
+    this.render();
+  }
+
+  setUpSearchBar() {
+    const stateAvg = "State Average";
+    const searchMenu = this.container.getElementsByClassName("menu")[0];
+    const counties = Object.keys(this.data).filter(c => c !== stateAvg).sort();
+    // make sure state average appears at the beginning of the search
+    counties.unshift(stateAvg);
+    counties.forEach(county => {
+      const element = document.createElement("div");
+      // state average should be bolded
+      element.className = county === stateAvg ? "item bold" : "item";
+      element.innerText = county;
+      searchMenu.appendChild(element);
+    });
+
+    const searchInput = this.container.getElementsByTagName("input")[0];
+    searchInput.addEventListener("change", e => {
+      const searchValue = e.target.value;
+      this.searchTerms = searchValue.split(";").filter(s => s !== "");
+      this.points.forEach(point => {
+        const searched = this.searchTerms.includes(point.county.toLowerCase());
+        searched ? point.onMouseEnter() : point.onMouseLeave();
```

Comment:
I'm sure you're aware considering your comment on the PR, but handling search with onMouseEnter and onMouseLeave isn't ideal since hovering over a searched county will undo the search.

In other words, if you first search for a county and then hover in and out of that searched county, you'd be calling onMouseLeave, which essentially removes the highlighting of the search.